xsens-tf-bridge/src/xsens-tf-bridge.cpp

```diff
@@ -0,0 +1,262 @@
+        segments[11] = "LeftShoulder";
+        segments[12] = "LeftUpperArm";
+        segments[13] = "LeftForeArm";
+        segments[14] = "LeftHand";
+        segments[15] = "RightUpperLeg";
+        segments[16] = "RightLowerLeg";
+        segments[17] = "RightFoot";
+        segments[18] = "RightToe";
+        segments[19] = "LeftUpperLeg";
+        segments[20] = "LeftLowerLeg";
+        segments[21] = "LeftFoot";
+        segments[22] = "LeftToe";
+        
+        i = 0;
+        segmentsCount = 49; //49
+        trueSegmentCount = 23;
+        tf.transforms.resize(49); //49
+        
+        for (i=0;i<segmentsCount;i++){
+        tf.transforms[i].header.frame_id = "ground";
+        }
+        
+        for (i=0;i<trueSegmentCount;i++){
+        tf.transforms[i].child_frame_id = segments[i];
+        }
+        
+        tf.transforms[23].child_frame_id = "L5_f1";
+        tf.transforms[24].child_frame_id = "L3_f1";
+        tf.transforms[25].child_frame_id = "T12_f1";
+        tf.transforms[26].child_frame_id = "T8_f1";
+        tf.transforms[27].child_frame_id = "T8_f2";
+        tf.transforms[28].child_frame_id = "Neck_f1";
+        tf.transforms[29].child_frame_id = "Neck_f2";
+        tf.transforms[30].child_frame_id = "Head_f1";
+        tf.transforms[31].child_frame_id = "RightUpperArm_f1";
+        tf.transforms[32].child_frame_id = "RightUpperArm_f2";
+        tf.transforms[33].child_frame_id = "RightForeArm_f1";
+        tf.transforms[34].child_frame_id = "RightHand_f1";
+        tf.transforms[35].child_frame_id = "LeftUpperArm_f1";
+        tf.transforms[36].child_frame_id = "LeftUpperArm_f2";
+        tf.transforms[37].child_frame_id = "LeftForeArm_f1";
+        tf.transforms[38].child_frame_id = "LeftHand_f1";
+        tf.transforms[39].child_frame_id = "RightUpperLeg_f1";
+        tf.transforms[40].child_frame_id = "RightUpperLeg_f2";
+        tf.transforms[41].child_frame_id = "RightLowerLeg_f1";
+        tf.transforms[42].child_frame_id = "RightFoot_f1";
+        tf.transforms[43].child_frame_id = "RightFoot_f2";
+        tf.transforms[44].child_frame_id = "LeftUpperLeg_f1";
+        tf.transforms[45].child_frame_id = "LeftUpperLeg_f2";
+        tf.transforms[46].child_frame_id = "LeftLowerLeg_f1";
+        tf.transforms[47].child_frame_id = "LeftFoot_f1";
+        tf.transforms[48].child_frame_id = "LeftFoot_f2";
+
+        handlerPort.open("/myModule");
+        attach(handlerPort);
+
+        driverServerName = "/xsens/cmd:i"; // >>yarp conf 10.255.36.7 10000 >> yarp rpc /xsens/cmd:i >>yarp read ... /xsens/frames:o
+        client_port.open("/XsensDriverService/client");
+//         if (!Network::connect("/XsensDriverService/client",driverServerName.c_str()))
+//         {
+//            std::cout << "Error! Could not connect to server " << driverServerName << std::endl;
+//            return -1;
+//         }
+        xsensDriver.yarp().attachAsClient(client_port);
+        cout << xsensDriver.segments() << endl;
```

Comment:
Getting the segments from the interface instead of hardcoding them is a good idea, but if you did not implemented you should remove this printouts.